oauth2-server-snap/lib/Network/OAuth2/Server/Snap.hs

```diff
@@ -64,6 +104,15 @@ createGrant
     :: Handler b (OAuth2 IO b) TokenGrant
 createGrant = return aGrant
 
+-- | Create an access token and send it to the client.
+createAndServeToken
```

Comment:
This should be called `createAndServeToken`, as it is exactly the same procedure for every single `grant_type`. It shouldn't be called directly though.